src/pages/resultsView/ResultsViewURLWrapper.ts

```diff
@@ -317,67 +325,37 @@ export default class ResultsViewURLWrapper
         }
     }
 
-    /**
-     * Query param clinicallist can be:
-     * - url encoded json object
-     * - comma seperated list (legacy)
-     * - undefined: clinical tracks not configured
-     * - 'null': all clinical tracks were deleted by user
-     *
-     * Ideally, we would like to simply give an empty string.
-     * The problem is that, in order to know whether to show
-     * some clinical tracks by default (such as "profiled-in",
-     * "samples per patient", etc), we need to know whether
-     * the clinical tracks have been updated by the user. If we
-     * pass an empty string, the router just deletes that parameter
-     * from the URL completely, making it indistinguishable from
-     * the initialization state. So we have to use 'null' here to
-     * distinguish the state of user having deleted all clinical tracks,
-     * because the alternative is to make a breaking change to the router library.
-     *
-     * @param clinicalTracks
-     */
-    public convertClinicalTracksToUrlParam(
-        clinicalTracks: ClinicalTrackConfig[]
-    ) {
-        if (clinicalTracks.length) {
-            return { clinicallist: JSON.stringify(clinicalTracks) };
-        } else {
-            return { clinicallist: ALL_TRACKS_DELETED };
+    private extractUserSettings(query: { [key: string]: any }) {
```

Comment:
could we do this outside of urlwrapper?

Reply:
so that urlWrapper is not aware at all of page settings information

Reply:
@alisman: After looking in the code, I would prefer to keep it in the url wrapper. 
These user settings are stored and retrieved from the url, just like the legacy `clinicallist` query param. See also `oncoprintSelectedClinicalTracks`. It is all url related functionality, so moving it to the `ResultViewPageStore` or `ResultViewOncoprint` doesn't feel logical to me (and whould further bloat those two already quite large files).